src/main/puzzle/editor/piece-chunk.gd

```diff
@@ -0,0 +1,31 @@
+class_name PieceControl
+extends LevelChunkControl
+"""
+A level editor chunk which contains several blocks which make up a single piece, such as a t-piece.
+"""
+
+enum EditorPiece {
+	PIECE_J, PIECE_L, PIECE_O, PIECE_P, PIECE_Q, PIECE_T, PIECE_U, PIECE_V,
+}
+
+export (EditorPiece) var _editor_piece: int
+
+var _editor_pieces := {
+	EditorPiece.PIECE_J: PieceTypes.piece_j,
+	EditorPiece.PIECE_L: PieceTypes.piece_l,
+	EditorPiece.PIECE_O: PieceTypes.piece_o,
+	EditorPiece.PIECE_P: PieceTypes.piece_p,
+	EditorPiece.PIECE_Q: PieceTypes.piece_q,
+	EditorPiece.PIECE_T: PieceTypes.piece_t,
+	EditorPiece.PIECE_U: PieceTypes.piece_u,
+	EditorPiece.PIECE_V: PieceTypes.piece_v,
+}
+
+var _orientation := 0
+
+func _refresh_tilemap() -> void:
```

Comment:
You actually got the code order right on the first try, for these 7 different components without looking it up? Wow, that's hard mode...